src/main/python/ttconv/imsc/elements.py

```diff
@@ -362,16 +362,20 @@ def from_model(
     `ctx` contains state information used in the process.
     '''
 
-    head_element = et.Element(HeadElement.qn)
+    head_element = None
 
     styling_element = StylingElement.from_model(ctx, model_doc)
 
     if styling_element is not None:
```

Comment:
There might be a more python-y way to fix this. Feel free to improve or request changes. 
The root cause was creating the head  and layout elements when they were not needed.

Reply:
Looks good. I have made a small addition to catch the corner case where an empty `styling` element would be added if a child `initial` element was not output due to an error.